app/commands/track/determine_concept_map_layout.rb

```diff
@@ -143,14 +128,21 @@ def adjacencies
       end
 
       def determine_nodes
-        track.concept_exercises.includes(:taught_concepts, :prerequisites).map.with_index do |exercise, idx|
-          Node.new(
-            index: idx,
-            slug: exercise.slug,
-            taught_concepts: exercise.taught_concepts.to_a,
-            prerequisites: exercise.prerequisites.to_a
-          )
-        end.freeze
+        nodes = track.concept_exercises.includes(:taught_concepts, :prerequisites).flat_map do |exercise|
+          exercise.taught_concepts.map do |concept|
+            Node.new(
+              index: nil,
+              slug: concept.slug,
+              name: concept.name,
+              prerequisites: exercise.prerequisites.pluck(:slug)
+            )
+          end
+        end
+
+        nodes.map.with_index do |node, index|
+          node.index = index
+          node
+        end
```

Comment:
I **think** this could maybe be:

```suggestion
        nodes.each.with_index { |node, index| node.index = index }
```

That would save creating another array identical to the first?